src/main/scala/beam/agentsim/agents/modalbehaviors/ChoosesMode.scala

```diff
@@ -627,8 +636,29 @@ trait ChoosesMode {
           routeHistory.rememberRoute(leg.travelPath.linkIds, leg.startTime)
         }
       }
+      // HACKY FIX: The old behavior just chose the first RH_TRANSIT itinerary from all the possible RH_TRANSIT
+      // itineraries produced by R5. But different itineraries may have different combinations of walk/rh for
+      // access/egress, and the egress RH legs in different itineraries may have different starting locations. So we'd
+      // really need to send separate RH requests for each RH_TRANSIT itinerary if we wanted to do it correctly. This
+      // sounds too complicated and costly, so instead we just run a route choice on the RH_TRANSIT itineraries returned
+      // by R5. These don't necessarily have correct wait times, but R5 has been updated to give them appropriate costs.
+      // Once we've chosen the best itinerary we can send requests to the RHM to fill in true costs and wait times
```

Comment:
If we generally have the same response from the router (except it may contain WALK as access or egress) then the rest of the `ChoosesMode` should be the same (except we don't need to send RH requests for WALK legs).

Reply:
I'm imagining a situation where the router returns two separate WALK->TRANSIT->RH itineraries, where the two egress RH legs pick up in different locations (e.g. at different transit stops). As I, understand it doing this would allow for only one RH request for the egress leg in this case, even though it could start from multiple locations. I'm not actually sure what the consequences of that would be (only having a maybe incorrect estimated wait time for the egress leg?), but it seems to me like we'd still be missing something.